OpenAPI/kiota/tutorials/dotnet-dependencyinjection.md

```diff
@@ -0,0 +1,288 @@
+          description: The error message
+        documentation_url:
+          type: string
+          description: The error documentation URL
+```
+
+You can then use the [Kiota](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool) command line tool to generate the API client classes. We are using the [deserializer](/openapi/kiota/using#--deserializer---ds) and [serializer](/openapi/kiota/using#--serializer---s) options to specify the serialization format, to only use JSON serialization.
+
+```bash
+kiota generate -l csharp -d github-releases.yml -c GitHubClient -n GitHub.ApiClient -o ./GitHub --deserializer Microsoft.Kiota.Serialization.Json.JsonParseNodeFactory --serializer Microsoft.Kiota.Serialization.Json.JsonSerializationWriterFactory
+```
+
+## Create extension methods
+
+Create a new file named _KiotaServiceCollectionExtensions.cs_ in the root of your project and add the following code.
+
+```csharp
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+/// <summary>
+/// Service collection extensions for Kiota handlers.
+/// </summary>
+public static class KiotaServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds the Kiota handlers to the service collection.
+    /// </summary>
+    /// <param name="services"><see cref="IServiceCollection"/> to add the services to</param>
+    /// <returns><see cref="IServiceCollection"/> as per convention</returns>
+    /// <remarks>The handlers are added to the http client by the <see cref="AttachKiotaHandlers(IHttpClientBuilder)"/> call, which requires them to be pre-registered in DI</remarks>
+    public static IServiceCollection AddKiotaHandlers(this IServiceCollection services)
+    {
+        services.AddTransient<UriReplacementHandler<UriReplacementHandlerOption>>();
+        services.AddTransient<RetryHandler>();
+        services.AddTransient<RedirectHandler>();
+        services.AddTransient<ParametersNameDecodingHandler>();
+        services.AddTransient<UserAgentHandler>();
+        services.AddTransient<HeadersInspectionHandler>();
+        return services;
+    }
+
+    /// <summary>
+    /// Adds the Kiota handlers to the http client builder.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Requires the handlers to be registered in DI by <see cref="AddKiotaHandlers(IServiceCollection)"/>.
+    /// The order in which the handlers are added is important, as it defines the order in which they will be executed.
+    /// <see href="https://github.com/microsoft/kiota-http-dotnet/blob/c1c295d3b0ebb2182b66d9a6858241117b59b540/src/KiotaClientFactory.cs#L50-L62">KiotaClientFactory.cs</see> for the default order.
+    /// </remarks>
+    public static IHttpClientBuilder AttachKiotaHandlers(this IHttpClientBuilder builder)
+    {
+        builder.AddHttpMessageHandler<UriReplacementHandler<UriReplacementHandlerOption>>();
```

Comment:
same here about using the factory?

Reply:
I literally have no clue how you could use an instance of something in a method that expects just the type in the generic specifier.

Reply:
See above